monai/transforms/inverse_transform.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Dict, Hashable, Optional, Tuple
+
+import numpy as np
+import torch
+
+from monai.transforms.transform import RandomizableTransform, Transform
+from monai.utils import optional_import
+
+sitk, has_sitk = optional_import("SimpleITK")
+vtk, has_vtk = optional_import("vtk")
+vtk_numpy_support, _ = optional_import("vtk.util.numpy_support")
+
+__all__ = ["InvertibleTransform", "NonRigidTransform"]
+
+
+class InvertibleTransform(Transform):
```

Comment:
would be great to have more documentation for clarity:
- the "undo stack" implementation is for each `key` of data?
-  the `info` in `data[key_transform]` should be compatible with the default collate?
- it's using `id()` so I guess it's not persistent, shall we support a persistent undo stack?
(I feel the method names could be `push_transforms`, `pop_transforms`...)